internal/updatecheck/update.go

```diff
@@ -39,59 +41,47 @@ var versionURL = githubVersionURL
 // CheckVersion returns a notification message to inform the user
 // about a new version of krew, or an empty string. The notification
 // will only be emitted once a day.
-func CheckVersion(basePath string) string {
+func CheckVersion(basePath string) (string, error) {
 	f := filepath.Join(basePath, "last_update_check")
-	lastCheck := loadTimestamp(f)
-	tag := fetchTag(lastCheck)
-	if version.GitTag() == tag {
-		return ""
-	}
-	saveTimestamp(f)
-	return color.New(color.Bold).Sprintf(upgradeNotification, version.GitTag(), tag)
-}
-
-// fetchTag tries to return the tag name of the most recent krew
-// release. If the last check happened recently, or an error occurs
-// the hardcoded tag of the executing krew binary will be returned.
-// This effectively disables the update notification message.
-func fetchTag(lastCheck time.Time) string {
-	if time.Since(lastCheck).Hours() <= 24 {
+	if lastCheck := loadTimestamp(f); time.Since(lastCheck).Hours() <= 24 {
 		klog.V(3).Info("Last check was recently, skipping update check")
-		return version.GitTag()
+		return "", nil
 	}
 
-	latestTag, err := fetchLatestTagFromGithub()
-	if err != nil || latestTag == "" {
-		klog.V(3).Infof("Could not fetch most recent tag name: %s", err)
-		return version.GitTag()
+	latestTag, err := fetchLatestTag()
+	if err != nil {
+		return "", errors.Wrapf(err, "could not determine latest tag")
 	}
+	saveTimestamp(f)
 
-	klog.V(4).Infof("Found latest tag %q", latestTag)
-	return latestTag
+	if version.GitTag() == latestTag {
```

Comment:
I think it would be nicer to actually use our semver pkg to do a `>` check here.

Reply:
or rather, maybe don't do the version tag here?

I feel like the `CheckVersion` isn't matching what this method does with reading last_update_check time + comparison with current version.

Reply:
I'd rather not do any semver parsing here. In tests, `version.GitTag` returns `unknown` which would not work.
Besides, we only allow the remote version to increase. So checking for equality also does the job.